client/galaxy/scripts/components/user/CustomBuilds.vue

```diff
@@ -0,0 +1,404 @@
+        .then(response => {
+          this.items = response.data;
+        })
+        .catch(error => {
+          console.log(error.response);
+        });
+    },
+
+    loadCustomBuildsMetadata (historyId) {
+      axios
+        .get(`${Galaxy.root}api/histories/${historyId}/custom_builds_metadata`)
+        .then(response => {
+          const fastaHdas = response.data.fasta_hdas;
+
+          if (fastaHdas.length > 0) {
+            this.fastaFiles = fastaHdas;
+            this.fastaFilesSelectDisabled = false;
+          }
+
+          this.fastaFilesLoading = false;
+          this.installedBuilds = response.data.installed_builds;
+        })
+        .catch(error => {
+          console.log(error.response);
+          this.fastaFilesLoading = false;
+        })
+    },
+
+    save (event) {
+      const data = {
+        id: this.form.id,
+        name: this.form.name,
+        "len|type": this.lengthType,
+        "len|value": this.lengthValue
+      };
+
+      if (!data.id.trim() || !data.name.trim() || !data["len|value"].trim()) {
+        this.showAlert("danger", "All inputs are required.");
+        return false;
+      }
+
+      axios
+        .put(`${this.customBuildsUrl}/${data.id}`, data)
+        .then(response => {
+          if (response.data.message) {
+            this.showAlert("warning", response.data.message);
+          } else {
+            this.showAlert("success", "Successfully added a new custom build.");
+          }
+          this.loadCustomBuilds();
+        })
+        .catch(error => {
+          const message = error.response.data.err_msg;
+          this.showAlert("danger", message || "Failed to create custom build.");
+          console.log(error.response);
+        });
+    },
+
+    deleteBuild (id) {
+      axios
+        .delete(`${this.customBuildsUrl}/${id}`)
+        .then(response => {
+          this.items = this.items.filter(i => i.id !== id);
+        })
+        .catch(error => {
+          const message = error.response.data.err_msg;
+          this.showAlert("danger", message || "Failed to delete custom build.");
+          console.log(error.response);
+        });
+    },
+
+    readFile (event) {
+      const file = event.target.files && event.target.files[0];
+
+      if (file) {
+        const reader = new FileReader();
+
+        reader.onprogress = e => {
+          if (e.lengthComputable) {
+            this.fileLoaded = Math.round(e.loaded / e.total) * 100;
+          }
+        };
+
+        reader.onload = () => {
+          this.fileLoaded = 0;
+          this.form.file = reader.result;
+        };
+
+        reader.readAsText(file);
+      }
+    },
+
+    showAlert (type, message) {
+      this.alertType = type;
+      this.alertMessage = message;
+      this.dismissCountDown = this.dismissSecs;
+    },
+
+    countDownChanged (dismissCountDown) {
+      this.dismissCountDown = dismissCountDown;
+    }
+  }
+}
+</script>
+
+<style>
+  .col {
+    padding-left: 5px;
```

Comment:
These spacing styles can be removed an we could use `pl-1` bootstrap styles instead.